### PR TITLE
feat(animation): Add animation polish

### PR DESCRIPTION
**Pull request by damassi**
### Description

This PR adds some light animation polish around the app as well as shows a possible animation organizational pattern to follow. Also a good amount more ux cleanup and code deletion: 

https://user-images.githubusercontent.com/236943/225479480-47e8d172-a120-4b37-8e0a-f3072b301fa3.mp4

cc @artsy/mobile-platform 

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -105,6 +105,7 @@ module.exports = {
     "no-undef": OFF,
     "no-useless-catch": OFF,
     "no-useless-escape": OFF,
+    "react-hooks/exhaustive-deps": OFF,
```
Comment (damassi) [inline .eslintrc.js:108]: This rule really is quite useless

File: src/utils/hooks/animations/useFadeInAnimation.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { useEffect } from "react"
```
Comment (damassi) [inline src/utils/hooks/animations/useFadeInAnimation.ts:1]: Note the folder - `hooks/animations`. For reusability.

File: src/utils/hooks/animations/useSlideUpFromBottomAnimation.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import { useEffect } from "react"
+import { Easing, useAnimatedStyle, useSharedValue, withTiming } from "react-native-reanimated"
+
+interface UseSlideUpFromBottomAnimationProps {
+  startAnimation: boolean | undefined
+}
+
+export const useSlideUpFromBottomAnimation = ({
```
Comment (damassi) [inline src/utils/hooks/animations/useSlideUpFromBottomAnimation.ts:8]: Then define the thing thats happening. Most things can be pretty generic.

File: src/components/Screen/BottomView.tsx [hunk 4]
```diff
@@ -56,78 +19,9 @@ export const BottomView: React.FC = ({ children }) => {
...
-  const insets = useSafeAreaInsets()
-
-  if (insets.bottom > 0) {
-    return null
-  }
-
-  return <Spacer y={2} />
-}
-
-const useKeyboard = (): [keyboardShowing: boolean, keyboardHeight: number] => {
```
Comment (damassi) [inline src/components/Screen/BottomView.tsx:91]: All of the changes in this file could be deleted by using KeyboardAvoidingView, though again not sure if i'm missing something subtle somewhere

File: src/screens/HomeTabs.tsx [hunk 5]
```diff
@@ -24,7 +24,7 @@ export const HomeTabs = () => {
         animated: false,
         leftElements: (
           <Touchable hitSlop={DEFAULT_HIT_SLOP} onPress={() => navigation.navigate("Settings")}>
-            <MenuIcon fill="onBackgroundHigh" />
+            <MenuIcon fill="onBackgroundHigh" top="1px" />
```
Comment (damassi) [inline src/screens/HomeTabs.tsx:27]: px perfect polish stuff 😅

File: src/screens/Search/Search.tsx [hunk 6]
```diff
@@ -69,12 +68,12 @@ export const Search = () => {
...
     <Screen>
-      <Screen.Body fullwidth safeArea={false}>
-        <Flex mx={2}>
+      <Screen.Body safeArea={false}>
+        <Flex>
           <Flex flexDirection="row" alignItems="center">
             <Input
               ref={searchInputRef}
-              style={{ borderWidth: 0 }}
+              style={{ borderWidth: 0, left: -11 }}
```
Comment (damassi) [inline src/screens/Search/Search.tsx:76]: ditto

File: src/components/Lists/DocumentList.tsx [hunk 7]
```diff
@@ -26,17 +26,19 @@ export const DocumentList: React.FC<DocumentListProps> = ({ documents }) => {
       numColumns={2}
       data={documents}
       renderItem={({ item, i }) => {
+        const documentItem = item as SelectedItemDocument
```
Comment (damassi) [inline src/components/Lists/DocumentList.tsx:29]: ditto here

File: src/components/Tabs/useListenForTabContentScroll.tsx [hunk 8]
```diff
@@ -42,7 +47,14 @@ const useAnimatedHeaderScrolling = (scrollY: SharedValue<number>) => {
...
 
-      setCurrScrollY(Math.floor(scrollY.value))
+      const prevTitleShown = prevScrollY >= NAVBAR_HEIGHT
+      const currTitleShown = animatedScrollY >= NAVBAR_HEIGHT
+
+      if (prevTitleShown === currTitleShown) {
+        return
+      }
+
+      runOnJS(update)(Math.floor(animatedScrollY))
```
Comment (damassi) [inline src/components/Tabs/useListenForTabContentScroll.tsx:57]: another fix from screen fallout. Need to run this stuff on another thread

**Comment by damassi** [comment]
Commented out the toast stuff for the time being while we wait for some eyes from design.

File: src/screens/Artwork/Artwork.tsx [hunk 9]
```diff
@@ -78,8 +78,12 @@ export const ArtworkPage: React.FC<{ slug: string }> = ({ slug }) => {
 
     navigation.navigate("AddItemsToAlbum", {
       artworkToAdd: artwork as SelectedItem,
-      closeBottomSheetModal: () => bottomSheetRef.current?.closeBottomSheetModal(),
     })
+
+    // Slight delay while screen transitions
+    // setTimeout(() => {
+    bottomSheetRef.current?.closeBottomSheetModal()
+    // }, 700)
```
Comment (brainbicycle) [inline src/screens/Artwork/Artwork.tsx:86]: do we no longer need the delay here?

File: src/screens/Albums/CreateOrEditAlbum/CreateOrEditAlbum.tsx [hunk 10]
```diff
@@ -87,16 +85,18 @@ export const CreateOrEditAlbum = () => {
...
+              //     title: `Successfully ${mode === "edit" ? "edited" : "created"} album.`,
+              //     type: "success",
+              //     onPress: () => {
+              //       console.log("hiii")
+              //     },
+              //   })
+              // })
             },
           })
-          closeBottomSheetModal?.()
```
Comment (brainbicycle) [inline src/screens/Albums/CreateOrEditAlbum/CreateOrEditAlbum.tsx:99]: do we not need this anymore?

Comment (damassi) [reply src/screens/Albums/CreateOrEditAlbum/CreateOrEditAlbum.tsx:99]: Nope, this was actually creating possible crashes on device as functions can't be passed along as route params (not serializable). Centralized most things inside the bottom-sheet actions.

File: src/screens/Artwork/Artwork.tsx (hunk 9 above)
Comment (damassi) [reply src/screens/Artwork/Artwork.tsx:86]: Ah this was me experimenting, lemme wrap with my new helper

Comment (damassi) [reply src/screens/Artwork/Artwork.tsx]: fixed, and wrapped with `waitForScreenTransition`

**Pull request merged**